app/src/main/java/com/example/songme/ui/home/HomeFragment.kt

```diff
@@ -5,16 +5,49 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-
+import android.widget.Toast
 import com.example.songme.R
+import com.example.songme.data.model.Track
+import com.example.songme.data.repository.TrackRepository
+import com.example.songme.data.source.remote.TrackRemoteDataSource
+import com.example.songme.ui.adapter.TrackAdapter
+import com.example.songme.ui.listener.OnItemClickListener
+import com.example.songme.utils.Constants.ERROR
+import kotlinx.android.synthetic.main.fragment_home.*
+
+class HomeFragment : Fragment(), HomeContentContract.View, OnItemClickListener {
 
-class HomeFragment : Fragment() {
+    private val trackList = ArrayList<Track>()
+    private val presenter: HomeContentContract.Presenter = HomeContentPresenter(
+        this,
+        TrackRepository.getInstance(TrackRemoteDataSource.getInstance())
+    )
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home, container, false)
+    ): View? =
+        inflater.inflate(R.layout.fragment_home, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        presenter.getTrending()
+    }
+
+    override fun showTracks(tracks: List<Track>) {
+        trackList.clear()
+        trackList.addAll(tracks)
+        val trackAdapter = TrackAdapter(tracks, this)
+        recyclerTrending?.apply {
+            adapter = trackAdapter
+        }
+    }
+
+    override fun showError(exception: Exception) {
+        Toast.makeText(context, ERROR + exception.toString(), Toast.LENGTH_SHORT).show()
```

Comment:
Viết extension để dùng lại nhiều lần hàm toast 
```suggestion
        Toast.makeText(context, exception.message.toString(), Toast.LENGTH_SHORT).show()
```